internal/core/hosts/host.go

```diff
@@ -0,0 +1,74 @@
+package hosts
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/trento-project/agent/internal/core/hosts/systemd"
+)
+
+type UnitInfo struct {
+	UnitFileState string `json:"unit_file_state"`
+}
+
+type SystemdUnitsStatus map[string]UnitInfo
+
+func DefaultUnitInfo(units ...string) SystemdUnitsStatus {
+	defaultInfo := make(SystemdUnitsStatus)
+	if len(units) == 0 {
+		units = []string{"pacemaker.service"}
+	}
+	for _, unit := range units {
+		defaultInfo[unit] = UnitInfo{
+			UnitFileState: "unknown",
+		}
+	}
+	return defaultInfo
+}
+
+func GetSystemdUnitsStatus(ctx context.Context, units ...string) SystemdUnitsStatus {
+	dbus, err := systemd.NewDbusConnector(ctx)
+	if err != nil {
+		slog.Error("Error while creating dbus connection", "error", err)
+	}
+	return getSystemdUnitsStatus(ctx, dbus, units...)
```

Comment:
nitpick: maybe we could have `return GetSystemdUnitsStatusWithCustomDbus(...` so it gives more sense of coverage, as the 2nd is explicitly tested